azurerm/resource_arm_metric_alertrule.go

```diff
@@ -18,10 +18,16 @@ func resourceArmMetricAlertRule() *schema.Resource {
 		Read:   resourceArmMetricAlertRuleRead,
 		Update: resourceArmMetricAlertRuleCreateUpdate,
 		Delete: resourceArmMetricAlertRuleDelete,
+
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
 
+		DeprecationMessage: `This resource has been renamed to 'azurerm_monitor_metric_alertrule' to match the other monitor resources and will be removed in v2.0. 
+
+Functionally these resources are the same and to upgrade all that will need to be done is replace 'azurerm_metric_alertrule' with 'azurerm_monitor_metric_alertrule'
+`,
```

Comment:
I've got a local branch with a guide and slightly different wording (for the other resources we're renaming):

```
DeprecationMessage: `The 'azurerm_autoscale_setting' resource is deprecated in favour of the renamed version 'azurerm_monitor_autoscale_setting'.

Information on migrating to the renamed resource can be found here: https://terraform.io/docs/providers/azurerm/guides/migrating-to-monitor-resources.html

As such the existing 'azurerm_autoscale_setting' resource is deprecated and will be removed in the next major version of the AzureRM Provider (2.0).
`,
```

rather than change the wording in this PR I'm going to merge this and then update this in my branch with the other resources